### PR TITLE
feat: editing visibility selection

### PR DESCRIPTION
**Pull request by Zeng1998**

### CODE REVIEW

**Comment by Zeng1998** [comment]
https://user-images.githubusercontent.com/31177490/200157122-7702250e-924f-4ad5-9773-0d1d2f78be48.mp4

File: web/src/components/MemoEditor.tsx [hunk 1]
```diff
@@ -404,6 +437,14 @@ const MemoEditor: React.FC = () => {
           <button className="action-btn" onClick={handleFullscreenBtnClick}>
             {state.fullscreen ? <Icon.Minimize className="icon-img" /> : <Icon.Maximize className="icon-img" />}
           </button>
+          <label className="form-label selector">
```
Comment (boojack) [inline web/src/components/MemoEditor.tsx:440]: I think it will cause overflow problem. Please take a look into the mobile style.

Comment (Zeng1998) [reply web/src/components/MemoEditor.tsx:440]: Yes, I ignored it. Any good suggestions? I don't know how to handle it.

For mobile style, how about placing the selector below the buttons?

Comment (boojack) [reply web/src/components/MemoEditor.tsx:440]: How about put the selector in the following place?
image

Comment (Zeng1998) [reply web/src/components/MemoEditor.tsx:440]: I tried two, but neither seemed perfect.

1.

creating:
image

The selector overwrited the text.

updating:
image


2.

creating:
image

A large space.

updating:
image

Comment (boojack) [reply web/src/components/MemoEditor.tsx:440]: I think we shouldn't show this when creating new memo as we already has a default visibility option. 

So let's show the visibility selector only when editing memo.

Comment (Jonnyan404) [reply web/src/components/MemoEditor.tsx:440]: > I tried two, but neither seemed perfect.

image
Not here?

**Comment by Zeng1998** [comment]
https://user-images.githubusercontent.com/31177490/200239342-c3039d61-67c9-457b-836f-71edfc7177d7.mp4

File: server/memo.go [hunk 2]
```diff
@@ -34,29 +34,6 @@ func (s *Server) registerMemoRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusBadRequest, "Memo content shouldn't be empty")
 		}
 
-		if memoCreate.Visibility == "" {
```
[the remark this answers is not in the document]
Comment (Zeng1998) [reply server/memo.go:37]: We don't receive a empty `visibility`, do we?

[the remark this answers is not in the document]
Comment (boojack) [reply server/memo.go:37]: We allow user to create memo without visibility in open API.

**Review by boojack — APPROVED** [review]
LGTM

**Pull request merged**